### PR TITLE
DEV-1169

### PR DESCRIPTION
**Pull request by JustinKim3420**
# Description

Provide a high-level description of changes performed in this PR

# Validation

Provide evidence that the changes you made are behaving as per the acceptance criteria. Please specify what tests you wrote to verify the functionality.

# Checklist

- [ ] I have performed manual testing to verify behavior of this PR and included evidence above
- [ ] Another person has reviewed, understood, and tested this work

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
-ENV tarballFileName="gomboc-ai-cli-$packageVersion.tgz"
-ENV tarFileName="gomboc-ai-cli-$packageVersion.tar"
-
 RUN npm install
-RUN npm run build
-RUN npm pack
-RUN gzip -d $tarballFileName
-RUN npm install -g $tarFileName
\ No newline at end of file
+RUN npm run build
```
Comment (jkamenik) [inline Dockerfile:7]: should be a single RUN command.

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
-RUN npm pack
-RUN gzip -d $tarballFileName
-RUN npm install -g $tarFileName
\ No newline at end of file
+RUN npm run build 
+
+# Prod modules
+FROM node:18.17.0-slim as prod-modules
+WORKDIR /app
+COPY --from=build /app ./
```
Comment (jkamenik) [inline Dockerfile:12]: Does `build` do something to `/app` that needs to be copied?  If so you should explain it in a comment.  If not, then this should be `COPY . .`

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+RUN apt-get -y update \
+    && apt-get -y install git \
+    && npm link 
+
+# Production
+FROM gcr.io/distroless/nodejs18-debian12
+WORKDIR /app
+
+# Application copy
+COPY --from=prod-modules /usr/bin /usr/bin
```
Comment (jkamenik) [inline Dockerfile:24]: If you are going to blindly copy the distro then at this point distroless is likely getting in the way.

The `apt-get -y update` in the previous stage is what is getting rid of the CVEs, so my recommendation is to get rid of distroless entirely and have `prod-modules` be the final container.

File: test/container-structure-test.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    path: '/app'
+    shouldExist: true
+    # TODO: Ideally we dont want to run as root
+    # permissions: '-rw-r--r--'
+    # uid: 1000
+    # gid: 1000
+    isExecutableBy: 'group'
+
+metadataTest:
+  entrypoint: ['/nodejs/bin/node']
```
Comment (jkamenik) [inline test/container-structure-test.yaml:14]: Add a commandTest to make sure the app at least runs.

```
commandTests:
  - name: "Smoke Test"
    command: "/nodejs/bin/node gomboc -v"
    exitCode: 0
    expectedOutput: ["Version: .*"]
```

**Review by jkamenik — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (JustinKim3420) [reply Dockerfile:24]: I would agree, however just using the image from prod-modules produces multiple high and critical CVEs so, even though its unclean, the final image produced by the distroless base images seems better.

**Review by jkamenik — DISMISSED** [review]

**Review by pepegc — APPROVED** [review]

**Pull request merged**